uploader/review/review.py

```diff
@@ -0,0 +1,283 @@
+                                 person_id=person.iperson.union_iperson.person_id)
+        cwpm.update_current_user_timestamp(request.user.username)
+        # cwpm.save()
+        person_maps.append(cwpm)
+
+    return person_maps
+
+
+def link_location_to_work(entities: QuerySet, relationship_type: str, union_work: CofkUnionWork, work_id, request) \
+        -> List[CofkWorkLocationMap]:
+    location_maps = []
+    for origin_or_dest in entities.filter(iwork_id=work_id).all():
+        cwlm = CofkWorkLocationMap(relationship_type=relationship_type,
+                                   work=union_work, location=origin_or_dest.location.union_location,
+                                   location_id=origin_or_dest.location.union_location.location_id)
+        cwlm.update_current_user_timestamp(request.user.username)
+        # cwlm.save()
+        location_maps.append(cwlm)
+
+    return location_maps
+
+
+def create_union_manifestations(work_id: int, union_work: CofkUnionWork, request, context) -> List[CofkManifInstMap]:
+    union_maps = []
+    for manif in context['manifestations'].filter(iwork_id=work_id).all():
+        union_dict = {'manifestation_creation_date_is_range': 0}
+        for field in [f for f in manif._meta.get_fields() if f.name != 'iwork_id']:
+            try:
+                CofkUnionManifestation._meta.get_field(field.name)
+                union_dict[field.name] = getattr(manif, field.name)
+
+            except FieldDoesNotExist:
+                # log.warning(f'Field {field} does not exist')
+                pass
+
+        union_manif = CofkUnionManifestation(**union_dict)
+        union_manif.work = union_work
+        union_manif.save()
+
+        if manif.repository_id is not None:
+            inst = context['institutions'].filter(id=manif.repository_id).first()
+            union_inst = CofkUnionInstitution.objects.filter(pk=inst.institution_id).first()
+
+            cmim = CofkManifInstMap(relationship_type=REL_TYPE_STORED_IN,
+                                    manif=union_manif, inst=union_inst, inst_id=union_inst.institution_id)
+            cmim.update_current_user_timestamp(request.user.username)
+            union_maps.append(cmim)
+            # cmim.save()
+
+    return union_maps
+
+
+def bulk_create(objects: List[Type[models.Model]]):
+    if objects:
+        try:
+            type(objects[0]).objects.bulk_create(objects, batch_size=500)
+        except IntegrityError as ie:
+            log.error(ie)
+            log.exception(ie)
+            # self.add_error(f'Could not create {type(objects[0])} objects in database.')
+
+
+def get_work(works, work_id) -> CofkCollectWork | None:
+    try:
+        work_id = int(work_id)
+    except ValueError:
+        return
+
+    try:
+        return [w for w in works  if w.id == work_id][0]
+    except IndexError:
+        pass
+
+
+def accept_work(request, context: dict, upload: CofkCollectUpload):
```

Comment:
consider to extract common method in `accept_work` and `accept_works`

Reply:
yes, I am a bit hesitant about this at the moment because of the different bulk_create operations used, I will wait a bit